compiler/src/iree/compiler/Codegen/Common/GPU/GPUTileSwizzleUtils.cpp

```diff
@@ -127,11 +136,113 @@ TileSwizzle getIntrinsicSwizzle(IREE::GPU::MMAIntrinsic intrinsic,
   // dimension each time.
   for (int i = layout.outer.size() - 1; i >= 0; --i) {
     if (layout.outer[i] != 1) {
-      unroll(swizzle, i, layout.outer[i]);
+      unroll(swizzle, i, layout.outer[i],
+             /*cross_thread=*/false, /*cross_intrinsic=*/false);
```

Comment:
I don't think this actually needs to be in reverse order, since there is at most one dim inserted at the start of each expanded group. It made it slightly harder to follow the logic here, since it seemed like the reverse order was relevant somehow.

Reply:
done (good catch!)